src/helpers.js

```diff
@@ -363,14 +367,28 @@ const constructInternalLink = (
   uid,
   pageCourse,
   useShortcodes,
-  isRelativeToRoot
+  isRelativeToRoot,
+  courseData
 ) => {
   const isSameCourse = linkCourse === pageCourse
   const strippedPath =
     pathRelativeToCourseRoot !== "/"
       ? stripSlashPrefix(pathRelativeToCourseRoot)
       : pathRelativeToCourseRoot
 
+  // if this links to an image resource, return the resource_file placeholder
+  const matchingFile = courseData["course_files"].filter(
```

Comment:
Can we use `find` instead of `filter` here, since we only care about the first element?